### DISCUSSION TITLE
Problems interpreting variable importances in multivariate time series forest

### DISCUSSION THREAD

**Discussion by flying-scotsman** [alan-turing-institute/sktime#736] [Q&A]
Hi sktime community!

I'm trying to understand variable importances in a multivariate time series forest. I've attached a diagram - it displays the 3 standard feature importances for a time series forest fitted on a dataset with 103 instances, 2 variables (here 1 & 2) and 28 time points. What's confusing me is how **they seem to be continuous at the boundary** - of course the importances for the last time point for variable 1 are directly before the first time point of variable 2. I've also observed that the ordering of the variables in the long time series affects the feature importances.

Can anyone explain to me what's happening here? **What I would expect: Discontinuous (but normalized) variable importance curves.**

Here's the code I'm using to extract the importances (df is a pandas.DataFrame with pandas.Series as cells and labels is a pandas.Series):

```python
X_train, X_test, y_train, y_test = train_test_split(df, labels, random_state=42)

steps = [
    ("concatenate", ColumnConcatenator()), 
    ("classify", TimeSeriesForestClassifier(n_estimators=200)),
]

clf = Pipeline(steps)
clf.fit(X_train, y_train)

importance = clf.steps[1][1].feature_importances_
importance['feature'] = numpy.repeat(list(X_train.columns), len(df.iat[0, 0]))
```

![103instances_2variables_28timepoints](https://user-images.githubusercontent.com/25385009/112120015-2a05c380-8bbe-11eb-98ac-fcbb247c8c48.png)
